File: src/main/java/com/lyaiya/softcode/Softcode.java
package com.lyaiya.softcode;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLInterModComms;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@Mod(
        modid = Tags.MOD_ID,
        dependencies = "required-after:forge@[14.23.5.2847,);required-after:mixinbooter@[8.6,)",
        useMetadata = true
)
public class Softcode {
    public static Logger LOGGER = LogManager.getLogger();

    @Mod.EventHandler
    public static void preInit(FMLPreInitializationEvent event) {
        LOGGER.debug("preInit");
    }

    @Mod.EventHandler
    public static void init(FMLInitializationEvent event) {
        LOGGER.debug("init");
    }

    @Mod.EventHandler
    public static void imcEvent(FMLInterModComms.IMCEvent event) {
        LOGGER.debug("imcEvent");
    }

    @Mod.EventHandler
    public static void postInit(FMLPostInitializationEvent event) {
        LOGGER.debug("postInit");
    }
}

File: src/main/java/com/lyaiya/softcode/Softcode.kt
package com.lyaiya.softcode

import com.lyaiya.softcode.constant.ModIdConstant
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLInterModComms
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

@Mod(
    modid = Softcode.MOD_ID,
    dependencies = "required-after:forge@[14.23.5.2847,);required-after:${ModIdConstant.FORGELIN_CONTINUOUS};required-after:${ModIdConstant.MIXIN_BOOTER}",
    modLanguageAdapter = "io.github.chaosunity.forgelin.KotlinAdapter",
    useMetadata = true
)
object Softcode {
    const val MOD_ID = "softcode"
    const val MOD_NAME = "Softcode"

    @JvmField
    val LOGGER: Logger = LogManager.getLogger()

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        LOGGER.info("preInit")
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        LOGGER.info("init")
    }

    @Mod.EventHandler
    fun imcEvent(event: FMLInterModComms.IMCEvent) {
        LOGGER.info("imcEvent")
    }

    @Mod.EventHandler
    fun postInit(event: FMLPostInitializationEvent) {
        LOGGER.info("postInit")
    }
}